auth0/resource_auth0_client.go

```diff
@@ -919,15 +947,103 @@ func flattenClientJwtConfiguration(jwt *management.ClientJWTConfiguration) []int
-		m["infinite_token_lifetime"] = refreshToken.InfiniteTokenLifetime
-		m["infinite_idle_token_lifetime"] = refreshToken.InfiniteIdleTokenLifetime
-		m["idle_token_lifetime"] = refreshToken.IdleTokenLifetime
+
+	m["rotation_type"] = refreshToken.RotationType
+	m["expiration_type"] = refreshToken.ExpirationType
+	m["leeway"] = refreshToken.Leeway
+	m["token_lifetime"] = refreshToken.TokenLifetime
+	m["infinite_token_lifetime"] = refreshToken.InfiniteTokenLifetime
+	m["infinite_idle_token_lifetime"] = refreshToken.InfiniteIdleTokenLifetime
+	m["idle_token_lifetime"] = refreshToken.IdleTokenLifetime
+
+	return []interface{}{m}
+}
+
+func flattenClientAddons(addons map[string]interface{}) []interface{} {
+	if addons == nil {
+		return nil
 	}
+
+	m := make(map[string]interface{})
+
+	if value, ok := addons["samlp"]; ok {
+		samlp := value.(map[string]interface{})
+
+		samlpMap := map[string]interface{}{
+			"audience":                           samlp["audience"],
+			"recipient":                          samlp["recipient"],
+			"mappings":                           samlp["mappings"],
+			"create_upn_claim":                   samlp["createUpnClaim"],
+			"passthrough_claims_with_no_mapping": samlp["passthroughClaimsWithNoMapping"],
+			"map_unknown_claims_as_is":           samlp["mapUnknownClaimsAsIs"],
+			"map_identities":                     samlp["mapIdentities"],
+			"signature_algorithm":                samlp["signatureAlgorithm"],
+			"digest_algorithm":                   samlp["digestAlgorithm"],
+			"destination":                        samlp["destination"],
+			"lifetime_in_seconds":                samlp["lifetimeInSeconds"],
+			"sign_response":                      samlp["signResponse"],
+			"name_identifier_format":             samlp["nameIdentifierFormat"],
+			"name_identifier_probes":             samlp["nameIdentifierProbes"],
+			"authn_context_class_ref":            samlp["authnContextClassRef"],
+			"typed_attributes":                   samlp["typedAttributes"],
+			"include_attribute_name_format":      samlp["includeAttributeNameFormat"],
+			"binding":                            samlp["binding"],
+			"signing_cert":                       samlp["signingCert"],
+			"logout":                             mapToState(samlp["logout"].(map[string]interface{})),
+		}
+
+		m["samlp"] = []interface{}{samlpMap}
+	}
+
+	for _, name := range []string{
+		"aws", "azure_blob", "azure_sb", "rms", "mscrm", "slack", "sentry",
+		"box", "cloudbees", "concur", "dropbox", "echosign", "egnyte",
+		"firebase", "newrelic", "office365", "salesforce", "salesforce_api",
+		"salesforce_sandbox_api", "layer", "sap_api", "sharepoint",
+		"springcm", "wams", "wsfed", "zendesk", "zoom",
+	} {
+		if value, ok := addons[name]; ok {
+			addonType := value.(map[string]interface{})
+			m[name] = mapToState(addonType)
+		}
+	}
+
+	return []interface{}{m}
+}
+
+func flattenClientMobile(mobile map[string]interface{}) []interface{} {
```

Comment:
To properly fix the surfaced error we needed to add a custom flatten of the client mobile.